Fails to build with OpenSSL 4.0.0

I received a Debian bug report, that xymon does not build with OpenSSL 4.0.0: https://bugs.debian.org/1139228

I can reproduce the issue with current GIT HEAD:
```
make[1]: Entering directory '/usr/local/src/Packages/xymon/xymon.roro/xymonnet'
gcc -g -O2 -Wall -Wno-unused -I/usr/include/tirpc -Wno-pointer-sign -D_REENTRANT -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -DLINUX -I/usr/local/src/Packages/xymon/xymon.roro/include  -DHAVE_OPENSSL  -c -o contest.o contest.c
contest.c: In function 'xymon_ASN1_UTCTIME':
contest.c:409:15: error: invalid use of incomplete typedef 'ASN1_UTCTIME' {aka 'struct asn1_string_st'}
  409 |         len=tm->length;
      |               ^~
contest.c:410:31: error: invalid use of incomplete typedef 'ASN1_UTCTIME' {aka 'struct asn1_string_st'}
  410 |         asn1_string=(char *)tm->data;
      |                               ^~
make[1]: *** [Makefile:51: contest.o] Error 1
make[1]: Leaving directory '/usr/local/src/Packages/xymon/xymon.roro/xymonnet'
```

--- a/xymonnet/contest.c
+++ b/xymonnet/contest.c
@@ -406,8 +406,13 @@ static char *xymon_ASN1_UTCTIME(ASN1_UTCTIME *tm)
 	int len, i;
 	int century=0,year=0,month=0,day=0,hour=0,minute=0,second=0;
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+	len=ASN1_STRING_length(tm);
+	asn1_string=(char *)ASN1_STRING_get0_data(tm);
+#else
 	len=tm->length;
 	asn1_string=(char *)tm->data;
+#endif
 
 	result[0] = '\0';
 	if (len < 10) return result;

--- a/xymonnet/httptest.h
+++ b/xymonnet/httptest.h
@@ -10,7 +10,7 @@
 /*                                                                            */
 /*----------------------------------------------------------------------------*/
 
-#ifndef __HTTPTESTT_H_
+#ifndef __HTTPTEST_H_
 #define __HTTPTEST_H_
 
 extern int  tcp_http_data_callback(unsigned char *buf, unsigned int len, void *priv);

